build/build-image/Dockerfile

```diff
@@ -14,15 +14,6 @@
 
 FROM golang:GO_VERSION
 
-# Avoid permission issues when glide pulls mercurial repos as root
-RUN printf "[trusted]\nusers = *\n" > /root/.hgrc
```

Comment:
@carolynvs  do you think dep is likely to use/need this?

Reply:
Both glide and dep are shelling out to git under the hood, so I'm guessing that both would need the same permission hacks.

Reply:
I'm inclined to take this out, and see if anything in travis fails. As far as I know we shouldn't be running any of the vendor modification stuff at build time. 

Any ideas if there's something I can do to dep to poke it and make it try and download a mercurial repo?

Reply:
You can run `dep ensure --add bitbucket.org/golang-dep/dep-test` at the root of the repo. That will cause dep to cache dep's test hg repo, and add it to Gopkg.toml.

Reply:
Looks like Travis didn't fail; do y'all think it's safe to merge this PR as is?

Reply:
I'm guessing it's cruft that we don't need to carry forward.